lib/reverse_sentence.rb

```diff
@@ -1,4 +1,60 @@
 # A method to reverse the words in a sentence, in place.
+
+# Share and explain the time and space complexities for your solution.
+# Time complexity:
```

Comment:
Because you're creating and using new arrays: word, word_indexes (aside: the plural of index is indices), the amount memory your algorithm will need to execute will be linearly proportional to the number of characters in the input string. As a result, the space complexity of your algorithm will be O(n).

From what I see, the time complexity of your solution will be O(n).

I'd encourage you to first reason and explain before sharing the final answer of the time and space complexities.

To see a different solution, and more importantly, see an explanation of the time and space complexities, take a look at https://github.com/Ada-C10/reverse_sentence/blob/solution/lib/reverse_sentence.rb